engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -929,46 +928,10 @@ applyItem (fromCStore, (iid, (itemFull, kit))) = do
 
 -- * AlterDir
 
--- | Ask for a direction and alter a tile in the specified way, if possible.
+-- | Ask for a direction and alter a tile, if possible.
 alterDirHuman :: MonadClientUI m
-              => [TriggerTile] -> m (FailOrCmd RequestTimed)
-alterDirHuman ts = do
-  UIOptions{uVi, uLeftHand} <- getsSession sUIOptions
-  let verb1 = case ts of
-        [] -> "alter"
-        tr : _ -> ttverb tr
-      keys = K.escKM
-             : K.leftButtonReleaseKM
-             : map (K.KM K.NoModifier) (K.dirAllKey uVi uLeftHand)
-      prompt = makePhrase
-        ["Where to", verb1 <> "? [movement key] [pointer]"]
-  promptAdd0 prompt
-  slides <- reportToSlideshow [K.escKM]
-  km <- getConfirms ColorFull keys slides
-  case K.key km of
-    K.LeftButtonRelease -> do
-      leader <- getLeaderUI
-      b <- getsState $ getActorBody leader
-      K.PointUI x y <- getsSession spointer
-      let (px, py) = (x `div` 2, y - K.mapStartY)
-          dir = Point px py `vectorToFrom` bpos b
-      if isUnit dir
-      then alterTile ts dir
-      else failWith "never mind"
-    _ ->
-      case K.handleDir uVi uLeftHand km of
-        Nothing -> failWith "never mind"
-        Just dir -> alterTile ts dir
-
--- | Try to alter a tile using a feature in the given direction.
-alterTile :: MonadClientUI m
-          => [TriggerTile] -> Vector -> m (FailOrCmd RequestTimed)
-alterTile ts dir = do
-  leader <- getLeaderUI
-  b <- getsState $ getActorBody leader
-  let tpos = bpos b `shift` dir
-      pText = compassText dir
-  alterTileAtPos ts tpos pText
+              => m (FailOrCmd RequestTimed)
+alterDirHuman = pickPoint "alter" >>= alterTileAtPos
```

Comment:
This is optional and subjective, but I'd case-analyze the `Maybe` already here, without passing it to `alterTileAtPos`, so that the latter has no `Maybe` in its type. Basically, usually it's best when functions have simple types and any plumbing to reconcile types of outputs of some functions to be fed to as inputs of others is done in glue code, not in the functions themselves. There are exceptions, of course, e.g., if `alterTileAtPos` was used many times and in each case with a `Maybe`.

Reply:
You're right, I focused bit too much on gathering all error cases in one place. Besides, now it may seem obvious why `Maybe` could be there, but I bet in a week I won't remember any of this :-)